www/js/lib/zimfile.js

```diff
@@ -20,7 +20,7 @@
  * along with Kiwix (file LICENSE-GPLv3.txt).  If not, see <http://www.gnu.org/licenses/>
  */
 'use strict';
-define(['xzdec_wrapper', 'util', 'utf8', 'q'], function(xz, util, utf8, Q) {
+define(['xzdec_wrapper', 'util', 'utf8', 'q', 'zimDirEntry'], function(xz, util, utf8, Q, zimDirEntry) {
```

Comment:
Not sure how to add a comment on the "blob" function. So am adding here.
This function took me the longest to figure out. Okay I didn't know what Promises were so that didn't help :)
But even after...I think what was missing to me was how the decompression process works. Just a brief note in the function documentation would be really useful. Like what is chunk size? Relationship to the compressed data read length. To read the 16 bytes to get blob address for example why is chunk size set to 5 KB. Basically some kind of note on how this works would be great. By changing these sizes (data length/chunk size) can reads be done faster?